server/app/services/mongodb/migrator.rb

```diff
@@ -64,17 +67,27 @@ def migrations
 
     def migrate
       ensure_indexes
-      with_dlock('mongodb_migrate', 60) do
+      release_stale_lock
+      with_dlock(LOCK_NAME, LOCK_TIMEOUT) do
         migrate_without_lock
       end
     end
 
+    def release_stale_lock
+      lock = DistributedLock.where(name: LOCK_NAME).first
+      if lock && lock.created_at < (LOCK_TIMEOUT * 3).seconds.ago
+        info "releasing stale distributed lock (locked at #{lock.created_at})"
+        lock.destroy
```

Comment:
Isn't this racey? With four puma workers, two of them release the lock...

```
W, [2017-04-10T10:45:28.929117 #28311]  WARN -- Mongodb::Migrator: releasing stale distributed lock (locked at 2017-04-10T07:42:25+00:00)
W, [2017-04-10T10:45:28.929168 #28309]  WARN -- Mongodb::Migrator: releasing stale distributed lock (locked at 2017-04-10T07:42:25+00:00)
```

AFAIK this race could potentially wipe out the actual active migrator lock?